lib/Driver/ToolChains.cpp

```diff
@@ -325,6 +325,12 @@ void ToolChain::addCommonFrontendArgs(const OutputInfo &OI,
     auto OptArg = inputArgs.getLastArgNoClaim(options::OPT_O_Group);
     if (!OptArg || OptArg->getOption().matches(options::OPT_Onone))
       arguments.push_back("-enable-anonymous-context-mangled-names");
+
+    // If the output file format is SIL, we will always print the debug info.
+    if (OI.CompilerOutputType == file_types::ID::TY_SIL) {
+      arguments.push_back("-Xllvm");
+      arguments.push_back("-sil-print-debuginfo");
```

Comment:
Do we also want to rename this option to -g (in a subsequent patch)?

Reply:
I did think about removing `-sil-print-debuginfo`. For example, adding a new boolean field in `SILOptions` for this purpose and set it up whenever `-g` is given. And the field will later be consumed by `SILPrinter`.
But then we lost the ability to print debug info into SIL file _without_ the help of driver -- for instance, directly using `swift-frontend -frontend ...`.  Also, IIUC, the new swift driver can only communicate with frontend via command line flags (please correct me if I'm wrong).
Or are you suggesting to create a _frontend_ flag named after `-g`?

Reply:
AFAICT `-g` is currently the frontend end flag for "emit debug info". I don't think we should also be passing a `-Xllvm` option.

Reply:
> AFAICT `-g` is currently the frontend end flag for "emit debug info". I don't think we should also be passing a `-Xllvm` option.

You're right about the `-g` flag. Then I'll just consolidate `-Xllvm -sil-print-debuginfo` into `-g` as suggested.